draft-ietf-tls-tls13.md

```diff
@@ -1566,7 +1566,9 @@ cipher_suites
   order of client preference. If the list contains cipher suites
   the server does not recognize, support, or wish to use, the server
   MUST ignore those cipher suites, and process the remaining ones as
-  usual. Values are defined in {{cipher-suites}}.
+  usual. Values are defined in {{cipher-suites}}. If the client is
+  attempting a PSK key establishment, it MUST only advertise the single
+  cipher suite associated with the PSK.
```

Comment:
"@ekr: from the spec: "Prior to accepting PSK key establishment, the server MUST validate the corresponding binder value (see Section 4.2.6.1 below). If this value is not present or does not validate, the server MUST abort the handshake."

this contradicts your scenario."

Hm.... I don't think so... Here's one order of operations.

1. Negotiate the cipher suite.
2. If a PSK is present, check to see that it has the same hash. If not, do a full handshake.
3. If a PSK is present and has the same hash, check the binder. If that fails, abort the handshake. If it succeeds, accept the PSK.

The key point here is that you don't get a mismatch in the binder unless something is really wrong (just like with the Finished MAC) and then you should just generate an error, not try to fall back to a non-PSK handshake.